src/PageState.hpp

```diff
@@ -48,6 +48,12 @@ struct PageState {
    */
   double circling_scale;
 
+  /**
+   * The last zoom mode on this page. This attribute is only used if
+   * PageSettings::distinct_zoom is enabled.
+   */
+  bool auto_zoom_enabled;
```

Comment:
Who initializes this?

Reply:
Its dynamically initialized by map_settings upon LeavePage(). I also added it to Clear() so its reset to a default value.